components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.html

```diff
@@ -0,0 +1,77 @@
+<div class="constraint-margin">
+  <div class="name-margin">
+    <label>
+      <span class="label">Name</span>
+    </label>
+    <div class="drop">
+      <mat-select panelClass="chef-dropdown" [(value)]="name_id" [disabled]="!constraintKeys.length">
+        <mat-option *ngFor="let list of constraintKeys" [value]="list" [ngClass]="{'checkmark':list==name_id}">
+          {{ list }}</mat-option>
+      </mat-select>
+    </div>
+  </div>
+  <div class="operator-margin">
+    <label>
+      <span class="label label-operator">Operator</span>
+    </label>
+    <div class="operator-dropdown">
+      <mat-select panelClass="chef-dropdown" [(value)]="operator_id" [disabled]="!constraintKeys.length">
+        <mat-option *ngFor="let list of operatorKeys" [value]="list" [ngClass]="{'checkmark':list==operator_id}">
+          {{ list }}</mat-option>
+      </mat-select>
+    </div>
+  </div>
+  <div class="version-margin">
+    <label>
+      <span class="label label-version">Version<span class="example-text">( example 1.2.3 )</span></span>
+    </label>
+    <form [formGroup]="secondFormGroup">
+      <chef-form-field>
+        <input chefInput formControlName="version" id="name-input-constraint" (keyup)="handleInput($event)"
+          autocomplete="off" />
+      </chef-form-field>
+    </form>
+  </div>
+  <div class="add">
+    <chef-button primary (click)="addRow()" [disabled]="!secondFormGroup.valid || !constraintKeys.length">Add
+    </chef-button>
+  </div>
+</div>
+<div class="fixed-height">
+  <div class="scroll">
+    <div class="row-margin" *ngFor="let dynamic of dynamicArray; let i = index;">
```

Comment:
Really need to give the reader an idea of what a dynamic or dynamicArray is; let's use better names.

Reply:
updated the variable name